cmd/csi-driver/csi-driver.go

```diff
@@ -51,6 +52,18 @@ var (
 			} else {
 				log.InitLogging(csiControllerLogFile, nil, true)
 			}
+			// Scanning for stale entries on each node where CSI resides, 
+			// this check makes sure that it runs only on hpe-csi-nodes pods 
+			podName := os.Getenv("POD_NAME")
+			if strings.Contains(podName, "hpe-csi-node-") { //&& !strings.Contains(podName, "hpe-csi-controller-") {
```

Comment:
You should be able to check the arguments if the driver is running in node or controller mode (`--node-service`)

Reply:
Updated